scripts/deploy.js

```diff
@@ -0,0 +1,33 @@
+const hre = require("hardhat");
+
+const ADMIN_ADDRESS = "0xc38c5f97B34E175FFd35407fc91a937300E33860" // Balancer Maxi LM Multisig on polygon
+const UPKEEP_CALLER_ADDRESS = "0x08a8eea76D2395807Ce7D1FC942382515469cCA1" // Chainlink Registry on polygon
```

Comment:
The chainlink registry now changes per launch, so this never works.  It would be better to set the UPKEEP_CALLER_ADDRESS to either the owner address or the deployer address until it is changed.  At least we know both those addresses can do something.

Would also be nice to move to supporting a list here in the contract of upkeep callers, and/or at bare min supporting the 0 address as the upkeep_caller which would allow anyone to call performUpkeep (unpermissioned)

Reply:
https://docs.chain.link/chainlink-automation/guides/forwarder

Reply:
my thought here is to have a setter for the `upkeepId`. this way the contract can determine the to be whitelisted forwarder programmatically. if the upkeep task is changed, the `upkeepId` should be changed (atomically).

Reply:
so I guess we can add the forward address as the upkeeper? I modified the contract now so upkeeper is a list of addresses like you proposed @Tritium-VLK .

Reply:
just be aware of the fact that the forwarder will be unique for every upkeep, even when moving it to a different owner or reinitialising it. the forwarder whitelist needs to be dynamic and setable